src/commands/deployWorkspaceProject/getDefaultValues/DefaultResourcesNameStep.ts

```diff
@@ -0,0 +1,139 @@
+            validateInput: this.validateInput,
+            asyncValidationTask: (name: string) => this.validateNameAvailability(context, name)
+        })).trim();
+
+        ext.outputChannel.appendLog(localize('usingResourceName', 'User provided the new resource name "{0}" as the default for resource creation.', resourceBaseName))
+
+        !context.resourceGroup && (context.newResourceGroupName = resourceBaseName);
+        !context.managedEnvironment && (context.newManagedEnvironmentName = resourceBaseName);
+        !context.registry && (context.newRegistryName = resourceBaseName.replace(/[^a-z0-9]+/g, ''));
+        !context.containerApp && (context.newContainerAppName = resourceBaseName);
+        context.imageName = `${resourceBaseName}:latest`;
+    }
+
+    public async configureBeforePrompt(context: DeployWorkspaceProjectContext): Promise<void> {
+        const workspaceName: string = cleanWorkspaceName(nonNullValueAndProp(context.rootFolder, 'name'));
+        if (this.validateInput(workspaceName) !== undefined) {
+            return;
+        }
+
+        if (!await this.isWorkspaceNameAvailable(context, workspaceName)) {
+            return;
+        }
+
+        if (!context.resourceGroup || !context.managedEnvironment || !context.registry || !context.containerApp) {
+            ext.outputChannel.appendLog(localize('usingWorkspaceName', 'Using workspace name "{0}" as the default for remaining resource creation.', workspaceName));
+        }
+
+        !context.resourceGroup && (context.newResourceGroupName = workspaceName);
+        !context.managedEnvironment && (context.newManagedEnvironmentName = workspaceName);
+        !context.registry && (context.newRegistryName = workspaceName.replace(/[^a-z0-9]+/g, ''));
+        !context.containerApp && (context.newContainerAppName = workspaceName);
+        context.imageName = `${context.containerApp?.name || workspaceName}:latest`;
+    }
+
+    public shouldPrompt(context: DeployWorkspaceProjectContext): boolean {
+        return (!context.resourceGroup && !context.newResourceGroupName) ||
+            (!context.managedEnvironment && !context.newManagedEnvironmentName) ||
+            (!context.registry && !context.newRegistryName) ||
+            (!context.containerApp && !context.newContainerAppName);
+    }
+
+    private validateInput(name: string | undefined): string | undefined {
+        name ??= '';
+
+        // No symbols are allowed for ACR - we will strip out any offending characters from the base name, but still need to ensure this version has an appropriate length
+        const nameWithoutSymbols: string = name.replace(/[^a-z0-9]+/g, '');
+        if (!validateUtils.isValidLength(nameWithoutSymbols, 5, 20)) {
+            return localize('invalidLength', 'The alphanumeric portion of the name should total to at least 5 characters while not exceeding 20 characters.');
```

Comment:
Feel free to suggest better wording for this**

Reply:
'The alphanumeric portion of the name must be least 5 characters but no more than 20 characters.'